src/Output.cpp

```diff
@@ -30,8 +36,16 @@ void compile_module_to_assembly(const Module &module, std::string filename)  {
 void compile_module_to_native(const Module &module,
                    std::string object_filename,
                    std::string assembly_filename) {
-    if (object_filename.empty()) object_filename = module.name() + ".o";
-    if (assembly_filename.empty()) assembly_filename = module.name() + ".s";
+    if (object_filename.empty()) {
+        if (module.target().os == Target::Windows) {
+            object_filename = module.name() + ".obj";
+        } else {
+            object_filename = module.name() + ".o";
+        }
+    }
+    if (assembly_filename.empty()) {
+        assembly_filename = module.name() + ".s";
```

Comment:
Windows convention is typically ".asm", though perhaps that depends wich assembler one uses. (I've seen ".nasm" as well.) Perhaps the .o vs. .obj distinction is for something other than visual convention however and .s can be left as is.